src/openrct2/ride/TrackDesign.cpp

```diff
@@ -277,10 +277,9 @@ rct_string_id TrackDesign::CreateTrackDesignTrack(const Ride& ride)
                 continue;
             }
 
-            int16_t x = location.x * 32;
-            int16_t y = location.y * 32;
+            CoordsXY mapLocation{ location.x * 32, location.y * 32 };
 
-            TileElement* tileElement = map_get_first_element_at(x >> 5, y >> 5);
+            TileElement* tileElement = map_get_first_element_at(location.x, location.y);
```

Comment:
This is wrong. It still needs the `/32`

Reply:
Does it? `location.x` and `location.y` were being multiplied by `32` and saved into `{x, y` just to then be shifted left by `5`, which takes away exactly the 32. I just started using them directly.

Reply:
ah right you are. I read that as `mapLocation` not `location`. This is fine.